### ISSUE TITLE
Pinniped Supervisor - interactively authenticate once per day/week/month

### ISSUE THREAD

**Issue by 0hlov3** [vmware-tanzu/pinniped#1004]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I've to copy the Auth-Code manually cause I'm a Safari User.

**Describe the solution you'd like**
I want a Configuration Option for the Supervisor Session Lifetime. It would be Nice, if we could Configure the Supervisor to ask once a week for interactively authentication.

**Describe alternatives you've considered**
if Safari would do the interactive authentication in the background, then the problem would not be a problem.

**Comment by enj** [comment]
Hi @0hlov3 thanks for opening this issue.

This feature is on our roadmap, but has security implications since longer lived credentials are inherently more risky.

A few things that need to be implemented first before we can offer this functionality:

1. Signing key rotation
2. mTLS binding of credentials
3. #866
4. #975 (I think this is the last item for the IDP refresh epic - these are already done: #340 #861 #862 #863 #881 #887)
5. ... I am probably forgetting something ...

All of these items are on our roadmap 😄

**Comment by enj** [comment]
A possible workaround would be to create a file called `open` and put it in `/some/other/bin`:

```bash
#!/bin/bash

/usr/bin/open -a '/Applications/Google Chrome.app' $*
```

And edit your kubeconfig

```yaml
...
      command: /full/path/to/pinniped
      env:
      - name: PATH
        value: /some/other/bin
      installHint: The pinniped CLI does not appear to be installed.  See https://get.pinniped.dev/cli
...
```

Basically this overrides the `open` binary that will be used by that particular kubeconfig by setting a custom `PATH`.  This allows you to use a different browser for pinniped logins without changing the default browser you use.  Of course you would have to be logged in to your IDP in the other browser (Chrome in this example), but that may be less inconvenient than manually copy-pasting.

**Comment by CYB3RV4D3R** [comment]
I would love to see this feature added for session length. For certain scenarios and for security I would like to limit the session to even say short as 10 mins for certain users. Will be following your roadmap and hope to see this feature added!

**Comment by 0hlov3** [comment]
> A possible workaround for Safari on macOS would be to create a file called `open` and put it in `/some/other/bin`:
> 
> ```shell
> #!/bin/bash
> 
> /usr/bin/open -a '/Applications/Google Chrome.app' $*
> ```
> 
> And edit your kubeconfig
> 
> ```yaml
> ...
>       command: /full/path/to/pinniped
>       env:
>       - name: PATH
>         value: /some/other/bin
>       installHint: The pinniped CLI does not appear to be installed.  See https://get.pinniped.dev/cli
> ...
> ```
> 
> Basically this overrides the `open` binary that will be used by that particular kubeconfig by setting a custom `PATH`. This allows you to use a different browser for pinniped logins without changing the default browser you use. Of course you would have to be logged in to your IDP in the other browser (Chrome in this example), but that may be less inconvenient than manually copy-pasting.

This solution works for us. Thank you! :)

**Comment by enj** [comment]
> I would love to see this feature added for session length. For certain scenarios and for security I would like to limit the session to even say short as 10 mins for certain users. Will be following your roadmap and hope to see this feature added!

@CYB3RV4D3R if you are using an OIDC IDP, you can usually configure the length of the refresh token.  Pinniped will honor this - if that refresh token stops working, the user's pinniped session will expire with it.

**Comment by JHansen2000** [comment]
@enj 
Hey, this request seems to have gone stale. Are there any updates on this outstanding request? 

It seems the only remaining issue on your initial list is #866. I am willing to contribute toward this feature if you can help me better define the remaining workload.
